tests/MongoDB.Driver.Core.Tests/Core/Operations/AsyncCursorTests.cs

```diff
@@ -180,16 +325,113 @@ public void CreateGetMoreCommand_should_return_expected_result_when_maxTime_is_p
         }
 
         [Fact]
-        public void Dispose_should_dispose_channel_source_when_cursor_id_is_zero()
+        public void CreateKillCursorsCommand_should_return_expected_result()
+        {
+            var subject = CreateSubject(cursorId: 1);
+
+            var result = subject.CreateKillCursorsCommand();
+
+            result.Should().Be("{ \"killCursors\" : \"test\", \"cursors\" : [NumberLong(1)] }");
+        }
+
+        [Fact]
+        public void Dispose_should_be_shielded_from_exceptions()
+        {
+            var mockChannelSource = new Mock<IChannelSource>();
+            mockChannelSource
+                .Setup(c => c.GetChannel(It.IsAny<CancellationToken>()))
+                .Throws<Exception>();
+
+            var subject = CreateSubject(cursorId: 1, channelSource: Optional.Create(mockChannelSource.Object));
+
+            Action action = () => subject.Dispose();
+            action.ShouldNotThrow<Exception>();
```

Comment:
You could just call subject.Dispose(); If it throws an exception the test will fail.

Reply:
Ok